frontend/src/components/utils/Authentication.tsx

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+
+const authenticatedRouts = ['/home'];
+const unauthenticatedRouts = ['/login', '/registration'];
+
+const Authentication = ({children}: any) => {
+    const history = useHistory();
+    const location = useLocation();
+
+    useEffect(() => {
+        let isAuthenticated = false;
+
+        const getUserRoute = () => {
+            let currentRoute;
+            let defaultRoute = '/login';
+
+            if(isAuthenticated) {
+                defaultRoute = '/home';
+                currentRoute = authenticatedRouts.find(route => route === location.pathname);
+            } else {
+                defaultRoute = '/login';
+                currentRoute = unauthenticatedRouts.find(route => route === location.pathname);
+            }
+
+            if(currentRoute) {
+                history.push(currentRoute);
+            } else {
+                history.push(defaultRoute);
+            }
+        }
+
+        try {
+            if(!localStorage.getItem('token')) {
+                isAuthenticated = false;
+                getUserRoute();
+            } else {
+                axios.get('http://localhost:3000/users').then(res => {
```

Comment:
There is no reason to query for all existing users for this. Just add a method in back-end that takes token and return user if that token is good. If token has expired return false, logout user from front-end and redirect them to `/login` page.